[BUG]sort_test.py::test_orderby_with_processing[Float] failing on dataproc/emr

**Describe the bug**
05:03:42  FAILED src/main/python/sort_test.py::test_orderby_with_processing[Float] - As...

```
05:03:42  _____________________ test_orderby_with_processing[Float] ______________________
05:03:42  
05:03:42  data_gen = Float
05:03:42  
05:03:42      @pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
05:03:42      def test_orderby_with_processing(data_gen):
05:03:42          assert_gpu_and_cpu_are_equal_collect(
05:03:42  >               lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')))
05:03:42  
05:03:42  src/main/python/sort_test.py:75: 
05:03:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
05:03:42  src/main/python/asserts.py:338: in assert_gpu_and_cpu_are_equal_collect
05:03:42      _assert_gpu_and_cpu_are_equal(func, True, conf=conf)
05:03:42  src/main/python/asserts.py:330: in _assert_gpu_and_cpu_are_equal
05:03:42      assert_equal(from_cpu, from_gpu)
05:03:42  src/main/python/asserts.py:92: in assert_equal
05:03:42      _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
05:03:42  src/main/python/asserts.py:40: in _assert_equal
05:03:42      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
05:03:42  src/main/python/asserts.py:33: in _assert_equal
05:03:42      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
05:03:42  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
05:03:42  
05:03:42  cpu = 1.2991000403417274e-05, gpu = 1.832030466175638e-05
05:03:42  float_check = <function get_float_check.<locals>.<lambda> at 0x7f52e36e4830>
05:03:42  path = [659, 'a']
05:03:42  
05:03:42      def _assert_equal(cpu, gpu, float_check, path):
05:03:42          t = type(cpu)
05:03:42          if (t is Row):
05:03:42              assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {}".format(path)
05:03:42              if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
05:03:42                  for field in cpu.__fields__:
05:03:42                      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
05:03:42              else:
05:03:42                  for index in range(len(cpu)):
05:03:42                      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
05:03:42          elif (t is list):
05:03:42              assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {}".format(path)
05:03:42              for index in range(len(cpu)):
05:03:42                  _assert_equal(cpu[index], gpu[index], float_check, path + [index])
05:03:42          elif (t is pytypes.GeneratorType):
05:03:42              index = 0
05:03:42              # generator has no zip :( so we have to do this the hard way
05:03:42              done = False
05:03:42              while not done:
05:03:42                  sub_cpu = None
05:03:42                  sub_gpu = None
05:03:42                  try:
05:03:42                      sub_cpu = next(cpu)
05:03:42                  except StopIteration:
05:03:42                      done = True
05:03:42      
05:03:42                  try:
05:03:42                      sub_gpu = next(gpu)
05:03:42                  except StopIteration:
05:03:42                      done = True
05:03:42      
05:03:42                  if done:
05:03:42                      assert sub_cpu == sub_gpu and sub_cpu == None, "CPU and GPU generators have different lengths at {}".format(path)
05:03:42                  else:
05:03:42                      _assert_equal(sub_cpu, sub_gpu, float_check, path + [index])
05:03:42      
05:03:42                  index = index + 1
05:03:42          elif (t is dict):
05:03:42              # TODO eventually we need to split this up so we can do the right thing for float/double
05:03:42              # values stored under the map some where, especially for NaNs
05:03:42              assert cpu == gpu, "GPU and CPU map values are different at {}".format(path)
05:03:42          elif (t is int):
05:03:42              assert cpu == gpu, "GPU and CPU int values are different at {}".format(path)
05:03:42          elif (t is float):
05:03:42              if (math.isnan(cpu)):
05:03:42                  assert math.isnan(gpu), "GPU and CPU float values are different at {}".format(path)
05:03:42              else:
05:03:42  >               assert float_check(cpu, gpu), "GPU and CPU float values are different {}".format(path)
05:03:42  E               AssertionError: GPU and CPU float values are different [659, 'a']
05:03:42  
05:03:42  src/main/python/asserts.py:74: AssertionError
05:03:42  ----------------------------- Captured stdout call -----------------------------
05:03:42  ### CPU RUN ###
05:03:42  ### GPU RUN ###
```

--- a/integration_tests/src/main/python/sort_test.py
+++ b/integration_tests/src/main/python/sort_test.py
@@ -72,10 +72,12 @@ def test_multi_orderby_with_limit(data_gen):
 @pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
 def test_orderby_with_processing(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
-            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')))
+            # avoid ambiguity in the order by statement for floating point by including a as a backup ordering column
+            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a'), f.col('a')))
 
 # We are not trying all possibilities, just doing a few with numbers so the query works.
 @pytest.mark.parametrize('data_gen', [byte_gen, long_gen, float_gen], ids=idfn)
 def test_orderby_with_processing_and_limit(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
-            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a')).limit(100))
+            # avoid ambiguity in the order by statement for floating point by including a as a backup ordering column
+            lambda spark : unary_op_df(spark, data_gen).orderBy(f.lit(100) - f.col('a'), f.col('a')).limit(100))

